editor/src/components/editor/store/dispatch.tsx

```diff
@@ -112,6 +112,11 @@ function processAction(
       ...working,
       userState: UPDATE_FNS.SET_SHORTCUT(action, working.userState),
     }
+  } else if (action.action === 'SET_CURRENT_THEME') {
+    return {
+      ...working,
+      userState: UPDATE_FNS.SET_CURRENT_THEME(action, working.userState),
+    }
```

Comment:
Ah, it seems I missed that action handlers had started creeping into here. Because there are so many of them, and we don't want to fill `dispatch.tsx` with noise, we call the action handlers here <https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/editor/store/editor-update.tsx#L48>. Please can you move this to there.

Also, or action handling code follows the pattern of always returning the updated `EditorState`, so please can you update the handler that you've added to `actions.tsx` to return the full `EditorState`.

Reply:
I moved the action handler here because the one in `editor-update.tsx` expects a return type of `EditorState`, which doesn't contain `UserState`, so this seemed to be the only option

Reply:
But maybe I'm missing something here?

Reply:
Sorry, I am a total idiot! I’m so unused to us updating anything other then EditorState via the actions that I completely forgot this isn’t even touching that!